src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -273,6 +281,22 @@ public void filterForExam(Exercise exercise, List<StudentParticipation> particip
                 if (result != null) {
                     participation.setResults(Set.of(result));
                 }
+                // for quiz exercises remove correct answer options / mappings from response
+                if (ZonedDateTime.now().isBefore(latestEndDate) && exercise instanceof QuizExercise) {
+                    QuizSubmission quizSubmission = (QuizSubmission) latestSubmission.get();
+                    quizSubmission.getSubmittedAnswers().forEach(submittedAnswer -> {
+                        QuizQuestion question = submittedAnswer.getQuizQuestion();
+                        if (question instanceof ShortAnswerQuestion) {
+                            ((ShortAnswerQuestion) question).setCorrectMappings(null);
```

Comment:
We need to remove also the solutions from short answer questions:
![image](https://user-images.githubusercontent.com/41108487/87862913-79bc2d00-c955-11ea-8099-c7e782a72a36.png)

Reply:
Not sure where exactly you saw this. I just tried it out and the solutions are not in the response any more